spinalcordtoolbox/deepseg/models.py

```diff
@@ -443,11 +461,16 @@ def install_model(name_model, custom_url=None):
     else:
         if not isinstance(url_field, dict):
             raise ValueError("Invalid url field in MODELS")
-        urls_used = {}
-        for seed_name, model_urls in url_field.items():
-            logger.info(f"\nInstalling '{seed_name}'...")
-            urls_used[seed_name] = download.install_data(model_urls,
-                                                         folder(os.path.join(name_model, seed_name)), keep=True)
+        # totalspineseg handles data downloading itself, so just pass the urls along
+        if name_model == 'totalspineseg':
+            tss_init.init_inference(data_path=Path(folder(name_model)), quiet=False, dict_urls=url_field)
```

Comment:
I noticed that the `init_inference` function behaves in a specific way:

- It [keeps the model `.zip` files on disk](https://github.com/neuropoly/totalspineseg/blob/88377f2d8bb60d73736496e30126c75511f830d9/totalspineseg/utils/install_weights.py#L96-L97) in the `exports/` folder (to avoid redownloading them)
- It will skip reinstalling the model [if the output folder already exists](https://github.com/neuropoly/totalspineseg/blob/88377f2d8bb60d73736496e30126c75511f830d9/totalspineseg/utils/install_weights.py#L91).

This is a bit troublesome for SCT because:

- [ ] It essentially stores 2 copies of the models on disk (taking up ~800MB of space instead of ~400MB)
- [ ] Users can't reinstall the models if something has happened to the files on disk, or if the versions have updated. (`sct_deepseg` has logic that will check the model urls used in the download, then reinstall the model if the URLs are not the latest version.)

Would you be open to changing `totalspineseg` to allow reinstallation and not storing the .zips on disk?

Reply:
I added a variable in the function `init_inference` to keep or not the zip after installation:
```python
def init_inference(
        data_path,
        dict_urls,
        store_export=True,
        quiet=False
```

Reply:
Thank you!!